mujoco_viewer/mujoco_viewer.py

```diff
@@ -941,17 +941,19 @@ def render_sensor_cameras_safe(self, if_on_scrn_as_overlay=False, if_segmentatio
                 camera.frame_stamp = self._gui_data.frame_stamp
                 
 
-    def acquire_sensor_camera_frames_safe(self):
+    def acquire_sensor_camera_frames_safe(self, write_to= None):
         camera_buffers = {"frame_buffer":{}, "depth_buffer":{}, "frame_stamp":{}}
         with self._camera_data_lock:
             for camera_name, camera in self._camera_data.items():
                 camera_buffers["depth_buffer"][camera_name] = copy.deepcopy(camera.depth_buffer)
                 camera_buffers["frame_buffer"][camera_name] = copy.deepcopy(camera.frame_buffer)
                 camera_buffers["frame_stamp"][camera_name] = copy.deepcopy(camera.frame_stamp)
-                # if write_to: [NOT-USED: Implementation to save frames directly]
+                # if write_to: #[NOT-USED: Implementation to save frames directly]
```

Comment:
uncomment these lines?

Reply:
- I moved the feature of saving camera-images to-disk from the viewer to the engine, since the viewer's job is to provide the camera images to the engine. The engine was anyways responsible for saving the streamed camera-images in a video; so this was an additional task that was not out-off  -scope for the engine.
- The saved image's names also have the time-stamp added on so as to get a unique image for each frame, and not just over-write an existing image.
- Enabled saving of gray-scale representation of the depth-camera, though it remains to be checked if the sensor-type being used is capable for perceiving depth-information.
- Also, I still kept the save-image-to-disk feature commented as it slows down the simulation. I don't think this feature is necessary anyways since I am able to save the video from the cameras, but I still want to keep a working script around as someone down the line can use it once we figure out how to avoid slowing down the simulation.